src/core/flow.js

```diff
@@ -498,6 +505,44 @@ export function Flow({ moves, phases, endIf, turn, events, plugins }) {
     return state;
   }
 
+  function UpdateStage(state, { arg, playerID }) {
+    if (!arg) return state;
+    if (typeof arg === 'string') {
+      arg = { stage: arg };
+    }
+
+    let { ctx } = state;
+    let {
+      activePlayers,
+      _activePlayersMoveLimit,
+      _activePlayersNumMoves,
+    } = ctx;
+
+    if (arg.stage) {
+      if (activePlayers === null) {
+        activePlayers = {};
+      }
+      activePlayers[playerID] = arg.stage;
```

Comment:
Should we check if this is a valid stage before letting the user enter it (similar to what we do for phases)?

Reply:
That’s not currently the case in `setActivePlayers`. I would suggest merging it as is and then tightening up things like that in a separate PR. (I can’t see any reason to _not_ ensure stages are valid.)

Reply:
OK sure.